ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/StaticQueryExecutor.java

```diff
@@ -659,20 +660,47 @@ private static TableRowsEntity proxyTo(
       final ConfiguredStatement<Query> statement,
       final ServiceContext serviceContext
   ) {
-    final RestResponse<KsqlEntityList> response = serviceContext
+    final RestResponse<List<StreamedRow>> response = serviceContext
         .getKsqlClient()
-        .makeKsqlRequest(owner.location(), statement.getStatementText());
+        .makeQueryRequest(owner.location(), statement.getStatementText());
 
     if (response.isErroneous()) {
       throw new KsqlServerException("Proxy attempt failed: " + response.getErrorMessage());
     }
 
-    final KsqlEntityList entities = response.getResponse();
-    if (entities.size() != 1) {
-      throw new RuntimeException("Boom - expected 1 entity, got: " + entities.size());
+    final List<StreamedRow> streamedRows = response.getResponse();
+    if (streamedRows.isEmpty()) {
+      throw new KsqlServerException("Invalid empty response from proxy call");
     }
 
-    return (TableRowsEntity) entities.get(0);
+    // Temporary code to convert from QueryStream to TableRowsEntity
+    // Tracked by: https://github.com/confluentinc/ksql/issues/3865
+    final Header header = streamedRows.get(0).getHeader()
```

Comment:
do we do this anywhere else in code? I am wondering if we should go out with temporary code. Whats your read on potential corner cases here? (aside from code cleanliness)

Whats the effort of also fixing 3865 properly? 1 day more? @apurvam I wdyt?

Reply:
1, maybe 2.

I agree this is starting to feel a little... risky.

Reply:
I will let @apurvam take the call.